src/metpy/calc/basic.py

```diff
@@ -101,7 +101,7 @@ def wind_direction(u, v, convention='from'):
 
     mask = wdir <= 0
     if np.any(mask):
-        wdir[mask] += units.Quantity(360., 'deg')
+        wdir[np.where(mask)] += units.Quantity(360., 'deg')
```

Comment:
This might work as `np.array` as well, which would be more consistent with the changes in `heat_index`.